tests/platform/test_reboot.py

```diff
@@ -21,42 +21,109 @@
 from check_interface_status import check_interface_status
 from check_transceiver_status import check_transceiver_basic
 from check_transceiver_status import all_transceivers_detected
+from psu_controller import psu_controller
 
+REBOOT_COMMAND = "command"
+REBOOT_CAUSE = "cause"
+REBOOT_TIMEOUT = "timeout"
+REBOOT_HELPER = "helper"
 
-def reboot_and_check(localhost, dut, interfaces, reboot_type="cold"):
+REBOOT_TYPE_WARMREBOOT = "warm"
+REBOOT_TYPE_COLDREBOOT = "cold"
+REBOOT_TYPE_FASTREBOOT = "fast"
+REBOOT_TYPE_REBOOT_BY_POWEROFF = "power off"
+REBOOT_TYPE_REBOOT_BY_WATCHDOG = "watchdog"
+
+reboot_ctrl_dict = {
+    REBOOT_TYPE_REBOOT_BY_POWEROFF : {
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "Power Loss"
+    },
+    REBOOT_TYPE_COLDREBOOT : {
+        REBOOT_COMMAND : "reboot",
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "reboot"
+    },
+    REBOOT_TYPE_FASTREBOOT : {
+        REBOOT_COMMAND : "fast-reboot",
+        REBOOT_TIMEOUT : 180,
+        REBOOT_CAUSE : "fast-reboot"
+    },
+    REBOOT_TYPE_WARMREBOOT : {
+        REBOOT_COMMAND : "warm-reboot",
+        REBOOT_TIMEOUT : 180,
+        REBOOT_CAUSE : "warm-reboot"
+    },
+    REBOOT_TYPE_REBOOT_BY_WATCHDOG : {
+        REBOOT_COMMAND : "python -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\"",
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "Watchdog"
+    }
+}
```

Comment:
I think these constants definitions are a little bit too much. Can we simplify like below:
```
REBOOT_TYPE_WARM = "warm"
REBOOT_TYPE_COLD = "cold"
REBOOT_TYPE_FAST = "fast"
REBOOT_TYPE_POWER_OFF = "power_off"
REBOOT_TYPE_WATCHDOG = "watchdog"

REBOOT_TYPES = {
    REBOOT_TYPE_POWER_OFF: {
        "timeout": 300
        "cause": "Power Loss" 
    },
    REBOOT_TYPE_COLD: {
        "command": "reboot",
        "timeout": 300,
        "cause": "reboot"
    },
    REBOOT_TYPE_FAST: {
        "command": "fast-reboot",
        "timeout": 180,
        "cause": "fast-reboot"
    },
    REBOOT_TYPE_WARM: {
        "command": "warm-reboot",
        "timeout": 180,
        "cause": "warm-reboot"
    },
    REBOOT_TYPE_WATCHDOG: {
        "command": "python -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\"",
        "timeout": 300,
        "cause": "watchdog"
    }
}
```

Reply:
you see that "command", "timeout", "cause" repeats for several times at different lines.
in this case, should we use a constant to replace it?
actually, I'm just not sure that what criteria should we follow...

Reply:
If use constans rather than hard coded strings, the constants have to appear multiple times at different lines too. What's more, we got extra lines defining the constants.
The "command", "timeout" and "cause" words are simple enough to be used as dictionary keys and not likely to change in the future. Just hard coding them in the REBOOT_TYPES dictionary as key names is simpler and more clear.

Reply:
After discussion, we agree that no need to use dictionary here.